src/main/java/bookbob/functions/FindVisit.java

```diff
@@ -0,0 +1,75 @@
+package bookbob.functions;
+
+import bookbob.entity.Patient;
+import bookbob.entity.Records;
+import bookbob.entity.Visit;
+
+import java.util.List;
+
+public class FindVisit {
+
+    //find visit by nric and print all visits to terminal
+    public static void findVisitByIc(String nric, Records records) {
+        List<Patient> patientList = records.getPatients();
+        boolean found = false;
+        for (Patient patient : patientList) {
+            if (patient.getNric().equals(nric)) {
+                List<Visit> visits = patient.getVisit();
+                found = true;
+
+                if (visits == null) {
```

Comment:
This condition will not be possible as visits is a mandatory field when creating a Patient objects. Hence, visits will always have at least a size of 1